web/src/engine/osk/src/input/gestures/specsForLayout.ts

```diff
@@ -315,8 +351,11 @@ export function flickEndContactModel(params: GestureParams): ContactModel {
           // Note:  if we wanted auto-triggering once the threshold distance were met,
           // we'd need to move its related logic into this method.
           return 'resolve';
-        } else if(calcLockedDistance(path.stats, baseStats.cardinalDirection as any) < params.flick.dirLockDist) {
-          return 'reject';
+        } else {
+          const { dir } = determineLockFromStats(baseStats);
+          if(calcLockedDistance(path.stats, dir) < params.flick.dirLockDist) {
+            return 'reject';
+          }
```

Comment:
There's no return for else?

Reply:
+1 -- currently our flags for compiling web don't call out missing return values and I think that's something we should be resolving in the future, because there is potential for bugs there:

https://github.com/keymanapp/keyman/blob/6cfcf08f3fad0cfdf389fea5da7e8dfdc989427e/web/tsconfig.base.json#L17

Reply:
Within this file in particular, and the way the gesture models are specified, it's quite intentional; `undefined` is a legitimate return type for the gesture models.

So, if/when the time comes, I'd just have to make them explicitly return `undefined` or `null`, I guess.

Reply:
Understood, it's just not clear that that's intentional so would encourage explicit return values everywhere.